docs/user-guide/response-codes-client-error.md

```diff
@@ -0,0 +1,22 @@
+# Styx Client Error (4xx) Response Codes
+
+When a client behaves in an unexpected way toward Styx, Styx will respond with specific error codes.
+This refers only to responses generated by Styx itself, not responses being proxied from downstream.
+For a general explanation of 4xx (Server Error) codes, please see the HTTP Spec.
+
+Note that interceptor plugins can also be programmed to modify response codes, so 
+they may introduce 4xx statuses for other reasons not listed here.
+
+
+## 400 Bad Request
+
+Occurs when a request is somehow incorrectly formatted. Examples:
+
+* Unrecognised HTTP verb
+* Multiple `Host` headers.
+
+
+## 408 Request Timeout
+
+Occurs if the channel from the client to Styx becomes idle, for example if the client starts sending a request
+but does not complete it.
```

Comment:
Please elaborate these aspects:

- More specifically, this can be thought of an "origin inactivity timeout". That is:
  - While origin is sending a HTTP Request, it must send something within `proxy.requestTimeoutMillis` (I believe this timeout can be configured for the admin interface too). Any inactivity exceeding this will trigger a 408 response code.
  - The timer applies only while HttpRequest is being transmitted "over the wire". That is, the timer is stopped when styx has fully received request. Therefore any delay or inactivity introduced by origin responding slowly will not trigger this timeout.

Reply:
At first, when you mention "origin" do you mean the client? This section is purely for client-errors and should have nothing to do with the origins.

Reply:
Hi yes. That is correct. My mistake.